src/main/java/seedu/address/logic/commands/CommandResult.java

```diff
@@ -21,6 +24,18 @@
      */
     private final boolean exit;
 
+    /**
+     * Constructs a {@code CommandResult} with the specified fields.
+     */
+    public CommandResult(String feedbackToUser, boolean isResidentCommand, boolean isRoomCommand,
+            boolean isResidentRoomCommand, boolean showHelp, boolean exit) {
+        this.feedbackToUser = requireNonNull(feedbackToUser);
+        this.isResidentCommand = isResidentCommand;
+        this.isRoomCommand = isRoomCommand;
```

Comment:
Maybe a enum like "CommandType" would make more sense as it is somewhat overcrowding the constructor. The check could be against the enum value instead if needed